spec/unit/util/package/version/rpm_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+require 'puppet/util/package/version/rpm'
+
+describe Puppet::Util::Package::Version::Rpm do
+
+  context "when parsing an invalid version" do
+    it "if is parsing symbols" do
+      expect { described_class.parse(:absent)}.to raise_error
+    end
+  end
+
+  context "when creating new version" do
+    it "is parsing basic version" do
+      v = described_class.parse('1:2.8.8-1.el6')
+      expect([v.epoch, v.version, v.release, v.arch ]).to match_array(['1', '2.8.8', '1.el6' , nil])
```

Comment:
I know i suggested this, but in this scenario it can be a bit tricky, `match_array` does not take into consideration order of the elements, so for example `epoch` could be `nil` and `arch` to be `'1'` and this test will pas.
It would be better to use `eq` here as it will also match the order of the elements